enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java

```diff
@@ -405,11 +382,49 @@ public boolean hasProcessExpired(OnboardingProcessEntity onboardingProcess) {
         // Check expiration for onboarding process due to process timeout
         final Duration processExpiration = onboardingConfig.getProcessExpirationTime();
         final Date createdDateExpirationProcess = DateUtil.convertExpirationToCreatedDate(processExpiration);
-        if (onboardingProcess.getTimestampCreated().before(createdDateExpirationProcess)) {
-            return true;
+        return onboardingProcess.getTimestampCreated().before(createdDateExpirationProcess);
+    }
+
+    private String lookupUser(final String processId, final Map<String, Object> identification) throws OnboardingProcessException {
+        try {
+            final LookupUserRequest lookupUserRequest = LookupUserRequest.builder()
+                    .identification(identification)
+                    .processId(processId)
+                    .build();
+            return onboardingProvider.lookupUser(lookupUserRequest).getUserId();
+        } catch (OnboardingProviderException e) {
+            throw new OnboardingProcessException("User lookup failed, error: " + e.getMessage(), e);
         }
+    }
 
-        return false;
+    @SneakyThrows(OnboardingProcessException.class)
```

Comment:
IMHO we exaggerate it with checked exceptions. Anyway, this Lombok annotation enables to use the checked exception in lambdas. Declaration at public level remains.